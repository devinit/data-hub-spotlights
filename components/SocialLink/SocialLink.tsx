import React from 'react';

const SocialLink = (props: any) => {
  return (
    <a href={props.url}>
      <img src={'/assets/svg/source/' + props.socialSource + '.svg'} alt={props.socialSource} />
      <style jsx>{`
        img {
          float: left;
          margin: 0 5px;
          height: 3em;
          width: 3em;
        }
      `}</style>
    </a>
  );
};

export { SocialLink };
