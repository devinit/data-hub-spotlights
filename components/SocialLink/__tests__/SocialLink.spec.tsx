/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SocialLink } from '../SocialLink';

describe('SocialLink', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<SocialLink socialSource="twitter" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
