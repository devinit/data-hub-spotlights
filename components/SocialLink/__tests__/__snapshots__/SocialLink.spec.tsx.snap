// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLink renders correctly 1`] = `
<a
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="twitter"
    src="/assets/svg/source/twitter.svg"
  />
  <style
    jsx={true}
  >
    
        img {
          float: left;
          margin: 0 5px;
          height: 3em;
          width: 3em;
        }
      
  </style>
</a>
`;
